// Copyright 2018 Istio Authors
//
//   Licensed under the Apache License, Version 2.0 (the "License");
//   you may not use this file except in compliance with the License.
//   You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the License is distributed on an "AS IS" BASIS,
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and
//   limitations under the License.

syntax = "proto3";

package envoy.security.v2;

import "google/protobuf/duration.proto";
import "validate/validate.proto";

message AltsSocket {
  // The location of a handshaker service, this is usually 169.254.169.254:8080
  // on GCE
  string handshaker_service = 1 [(validate.rules).string.min_bytes = 1];

  // The acceptable service accounts from peer, peers not in the list will be
  // rejected in the handshake validation step.
  // If empty, no validation will be performed.
  repeated string peer_service_accounts = 2;
}
